custom_navbutton_completion.js

```diff
@@ -0,0 +1,119 @@
+/**
```

Comment:
I'd be tempted to drop Moodle 2.2 - 2.4 compatibility here and package this up as a YUI module (or even consider dropping 2.2-2.8 compatibility and making it an AMD module instead).